device/thunder_ripple_sdk/src/client/thunder_client.rs

```diff
@@ -51,18 +53,108 @@ use ripple_sdk::{
     utils::error::RippleError,
 };
 use serde::{Deserialize, Serialize};
+use tokio::sync::oneshot::Sender;
 use url::Url;
 
-use crate::thunder_state::ThunderConnectionState;
-use crate::utils::get_error_value;
-
+use super::thunder_async_client::{ThunderAsyncClient, ThunderAsyncRequest, ThunderAsyncResponse};
 use super::thunder_client_pool::ThunderPoolCommand;
+use super::thunderbroker_plugins_status_mgr::{BrokerCallback, BrokerSender};
 use super::{
     jsonrpc_method_locator::JsonRpcMethodLocator,
     plugin_manager::{PluginActivatedResult, PluginManagerCommand},
 };
+use ripple_sdk::api::device::device_operator::{DeviceChannelRequest, DeviceResponseSubscription};
+use ripple_sdk::tokio::sync::mpsc::Receiver;
+use std::sync::RwLock;
 use std::{env, process::Command};
 
+pub type BrokerSubMap = HashMap<String, DeviceResponseSubscription>;
+pub type BrokerCallbackMap = HashMap<u64, Option<OneShotSender<DeviceResponseMessage>>>;
+
+#[derive(Debug)]
+pub struct ThunderClientManager;
+
+impl ThunderClientManager {
+    fn manage(
+        client: ThunderClient,
+        request_tr: Receiver<ThunderAsyncRequest>,
+        mut response_tr: Receiver<ThunderAsyncResponse>,
+        thndr_endpoint_url: String,
+    ) {
+        let client_c = client.clone();
+
+        tokio::spawn(async move {
+            if let Some(thndr_asynclient) = &client_c.thndr_asynclient {
+                println!("@@@NNA----> from thunder_client to async start call....");
+                thndr_asynclient
+                    .start(&thndr_endpoint_url, request_tr)
+                    .await;
+            }
+            error!("Thunder disconnected so reconnecting");
+        });
+        /*thunder async response will get here */
+        tokio::spawn(async move {
+            println!("@@@NNA----> in thunder::manage response block entrance...");
+            while let Some(response) = response_tr.recv().await {
+                println!(
+                    "@@@NNA...manage fn response block..with response{:?}",
+                    response
+                );
+                if let Some(id) = response.get_id() {
+                    println!("@@@NNA...manage fn response id: {:?}", id);
+                    if let Some(cb) = client.clone().broker_callbacks {
```

Comment:
`cb` doesn't hold a single callback it is a map of callbacks, so rename to `broker_callbacks`.